source/exe/process_wide.cc

```diff
@@ -58,7 +57,12 @@ ProcessWide::~ProcessWide() {
 
   ASSERT(init_data.count_ > 0);
   if (--init_data.count_ == 0) {
-    ares_library_cleanup();
+    // The c-ares library init is done in a thread-safe way when it is actually used for resolving.
+    // Clean it up here in case it is ever used.
+    if (auto* dns_factory = Config::Utility::getAndCheckFactoryByName<Network::DnsResolverFactory>(
+            std::string(Network::CaresDnsResolver), true)) {
+      dns_factory->terminate();
```

Comment:
I think this should be done by category vs. specific resolver?

Reply:
Yes that is correct. The way you wrote the code is that you generically call initialize() on the factory, not calling terminate() on the factory doesn't make sense to me so let's go ahead and do that now. Thank you!

Reply:
I see what you mean now. Yeah, that make sense. We should do same thing for initialize and terminate.

Since what we want to do is specific for c-ares factory. I added a c-ares check there before initialize. This make them consistent.  PTAL.